pyccl/pk2d.py

```diff
@@ -56,10 +57,8 @@ class Pk2D(object):
              logarithm of the power spectrum. Otherwise, the true value
              of the power spectrum is expected. Note that arrays will be
              interpolated in log space if `is_logp` is set to `True`.
-        cosmo (:class:`~pyccl.core.Cosmology`): Cosmology object. The cosmology
-             object is needed in order if `pkfunc` is not `None`. The object is
-             used to determine the sampling rate in scale factor and
-             wavenumber.
+        cosmo (:class:`~pyccl.core.Cosmology`, optional): Cosmology object.
+             Used to determine sampling rates in scale factor and wavenumber.
```

Comment:
Is this still the case? (I.e. is this still used to determine sampling rates?)

Reply:
Can you add "Will be deprecated in the next CCL version"?

Reply:
Let's not deprecate this. `cosmo` is still used to extrapolate backwards in time using the linear power spectrum, when the requested scale factor is outside of the range of the `Pk2D` object.
With my changes, `eval` will work as before, but users won't be able to extrapolate backwards.

Reply:
Oh wait no! I got my PRs mixed up. No, we won't be deprecating this. `cosmo` will stay there, but I think I've addressed your point in another PR of mine, #923 .